lib/dialects/mysql/query/compiler.js

```diff
@@ -26,7 +26,13 @@ class QueryCompiler_MySQL extends QueryCompiler {
 
     const { ignore, merge, insert } = this.single;
     if (ignore) sql = sql.replace('insert into', 'insert ignore into');
-    if (merge) sql += this._merge(merge.updates, insert);
+    if (merge) {
+      sql += this._merge(merge.updates, insert);
+      const wheres = this.where();
+      if (wheres) {
```

Comment:
It feels like there might be a better way to check whether `where` is called after a `merge` statement for MySQL, but I could not figure out what it is